building from source failed

I downloaded, git submodule init, cmake ., make and it spat this out:

[ 27%] Built target LDCShared
make[2]: *** No rule to make target 'lib/libldmd.a', needed by 'bin/ldmd2'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:126: CMakeFiles/ldmd2.dir/all] Error 2

did i miss something or is the makefile broken?

> I downloaded, git submodule init, cmake ., make and it spat this out:
> 
> [ 27%] Built target LDCShared
> make[2]: *** No rule to make target 'lib/libldmd.a', needed by 'bin/ldmd2'. Stop.
> make[1]: *** [CMakeFiles/Makefile2:126: CMakeFiles/ldmd2.dir/all] Error 2
> 
> did i miss something or is the makefile broken?

I rolled back to v1.17.0 
1.18.0-beta2 does not work build with LLVM9

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -644,7 +644,7 @@ else()
         "${LDC_EXE_FULL}"
         "${LDC_D_SOURCE_FILES}"
         ""
-        "$<TARGET_LINKER_FILE:${LDC_LIB}>;${LDC_LINKERFLAG_LIST}"
+        "${LDC_LINKERFLAG_LIST}"
         "${FE_RES}"
         "${LDC_LIB}"
         ${COMPILE_D_MODULES_SEPARATELY}
@@ -687,7 +687,7 @@ build_d_executable(
     "${LDMD_EXE_FULL}"
     "${LDMD_D_SOURCE_FILES}"
     ""
-    "$<TARGET_LINKER_FILE:LDMD_CXX_LIB>;${LDC_LINKERFLAG_LIST}"
+    "${LDC_LINKERFLAG_LIST}"
     ""
     "LDMD_CXX_LIB"
     ${COMPILE_D_MODULES_SEPARATELY}
@@ -846,7 +846,7 @@ build_d_executable(
     "${LDC_UNITTEST_EXE_FULL}"
     "${LDC_D_SOURCE_FILES}"
     "-unittest"
-    "$<TARGET_LINKER_FILE:${LDC_LIB}>;${LDC_LINKERFLAG_LIST}"
+    "${LDC_LINKERFLAG_LIST}"
     ""
     "${LDC_LIB}"
     ON # always build separately (faster with parallelization)

--- a/cmake/Modules/BuildDExecutable.cmake
+++ b/cmake/Modules/BuildDExecutable.cmake
@@ -61,7 +61,7 @@ function(build_d_executable target_name output_exe d_src_files compiler_args lin
             RUNTIME_OUTPUT_DIRECTORY ${output_dir}
             LINKER_LANGUAGE          CXX
         )
-        target_link_libraries(${target_name} ${linker_args} ${D_LINKER_ARGS})
+        target_link_libraries(${target_name} ${link_deps} ${linker_args} ${D_LINKER_ARGS})
     else()
         # Use a response file on Windows when compiling separately, in order not to
         # exceed the max command-line length.
@@ -72,10 +72,16 @@ function(build_d_executable target_name output_exe d_src_files compiler_args lin
             set(objects_args "@${output_exe}.rsp")
         endif()
 
+        set(dep_libs "")
+        foreach(l ${link_deps})
+            list(APPEND dep_libs "-L$<TARGET_LINKER_FILE:${l}>")
+        endforeach()
+
         translate_linker_args(linker_args translated_linker_args)
+
         add_custom_command(
             OUTPUT ${output_exe}
-            COMMAND ${D_COMPILER} ${dflags} ${DDMD_LFLAGS} -of${output_exe} ${objects_args} ${translated_linker_args}
+            COMMAND ${D_COMPILER} ${dflags} ${DDMD_LFLAGS} -of${output_exe} ${objects_args} ${dep_libs} ${translated_linker_args}
             WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
             DEPENDS ${object_files} ${link_deps}
         )

